cpp/include/cudf/aggregation.hpp

```diff
@@ -148,6 +148,28 @@ class groupby_scan_aggregation : public virtual aggregation {
   groupby_scan_aggregation() {}
 };
 
+/**
+ * @brief Derived class intended for reduction usage.
+ */
+class reduce_aggregation : public virtual aggregation {
+ public:
+  ~reduce_aggregation() override = default;
+
+ protected:
+  reduce_aggregation() {}
```

Comment:
Remind me why these are necessary?

Reply:
The base `aggregation` constructor is deleted so these need to be here.

Reply:
Why can't `reduce_aggregation` constructor also be deleted?

Reply:
It gets called by the constructors of the derived classes.  `max_aggregation`, `sum_aggregation`, etc.